include/pagmo/algorithms/gaco.hpp

```diff
@@ -0,0 +1,974 @@
+                // In case a value for the focus parameter (different than zero) is passed, this limits the maximum
+                // value of the standard deviation
+                sigma_vec[h - 1] = (ub[h - 1] - lb[h - 1]) / m_focus;
+            } else {
+                sigma_vec[h - 1] = (d_max - d_min) / gen_mark;
+            }
+        }
+    }
+
+    /**
+     * Function which generates new individuals (i.e., ants)
+     *
+     * @param[in] pop Population: the current population of individuals is passed
+     * @param[in] prob_cumulative Cumulative probability vector: this vector determines the probability for choosing the
+     * pdf to generate new individuals
+     * @param[in] omega Omega: one of the three pheromone values. These are the weights which are used in the
+     * multi-kernel gaussian probability distribution
+     * @param[in] sigma Sigma: one of the three pheromone values. These are the standard deviations which are used in
+     * the multi-kernel gaussian probability distribution
+     * @param[in] X_new New ants: in this vector the new ants which will be generated are stored
+     * @param[in] gen Current number of generation: this represents the current generation number
+     * @param[in] sol_archive Solution archive: the solution archive is useful for retrieving the current individuals
+     * (which will be the means of the new pdf)
+     */
+    void generate_new_ants(const population &pop, std::uniform_real_distribution<> dist, vector_double prob_cumulative,
+                           vector_double omega, vector_double sigma, std::vector<vector_double> &dvs_new, double gen,
+                           std::vector<vector_double> &sol_archive) const
+    {
+
+        const auto &prob = pop.get_problem();
+        auto pop_size = pop.size();
+        auto n_con = prob.get_nx();
+        const auto bounds = prob.get_bounds();
+        const auto &lb = bounds.first;
+        const auto &ub = bounds.second;
+        vector_double fitness_old;
+        vector_double fitness_new;
+
+        // I hereby generate the new ants based on a multi-kernel gaussian probability density function. In particular,
+        // I select one of the pdfs that make up the multi-kernel, by using the probability stored in the prob_cumulative
+        // vector. A multi-kernel pdf is a weighted sum of several gaussian pdf.
+
+        if (m_omega_strategy == 1) {
+            for (decltype(pop_size) j = 0u; j < pop_size; ++j) {
+                vector_double dvs_new_j(
+                    n_con); // here I store all the variables associated with the j_th element of the sol_archive
+
+                for (decltype(n_con) h = 0u; h < n_con; ++h) {
+                    double g_h = 0;
+
+                    for (decltype(sol_archive.size()) k = 0u; k < sol_archive.size(); ++k) {
+                        std::normal_distribution<double> gauss_pdf_1{sol_archive[k][1 + h], sigma[h]};
```

Comment:
I see here (and later) you are instantiating  several ```std::normal_distribution<double>``` changing the std and mean. I suggest to instantiate only one of these objects (maybe above where you instantiate the uniform_real) ...
```
std::normal_distribution<double> gauss{0., 1.};
```
then you can draw a number anywhere in the code, with generic mu and std as follows:
```
my_normally_distributed_number = mu + std * gauss(m_e);
```
this produces code that is both more elegant and more efficient.